lib/client/bundler/proto.js

```diff
@@ -66,145 +125,250 @@ module.exports = function(ss, bundlers, bundlerById, options) {
+   * Provides the view and the pack functions with a
+   * list of entries for an asset type relative to the client directory.
+   * The default implementation is used.
+   *
+   * An entry describes:
+   * - file: source file name relative to client directory
+   * - importedFrom: Pulled in by require in other file
+   * - content: If pre-loaded
+   * - type: If system asset it will have type lib/mod
+   * - bundle: which production bundle (html, tmpl, css, js, worker)
+   * - assetType: css, js, html
+   * - extension: source file extension
+   *
+   * @param {String} assetType js/css/tmpl
+   * @param {Object} systemAssets Collection of libs, modules, initCode
+   * @returns {[AssetEntry]} List of output entries
+   */
+  function entries(assetType,systemAssets) {
+    // jshint validthis:true
+    var includes = this.client.includes;
+    return ss.bundler.entries(this.client, assetType, systemAssets).filter(function(entry) {
+      return entry.includeType? includes[entry.includeType] : true;
+    });
+  }
 
-      // mixin system constants
-      for(k in system.assets.constants) {
-        result[k] = system.assets.constants[k].value;
-      }
+  /**
+   * @ngdoc method
+   * @name bundler.default:default#system
+   * @methodOf bundler.default:default
+   * @function
+   * @description
+   * Return entries for the loader/start/module depending on includes.system client config and system assets.
+   * Multiple arguments will attempt to add module for the name.
+   * Special module names are 'loader' and 'start'.
+   *
+   * @returns {AssetEntry} Resource entries array
+   */
+  function systemModule(/* first_name */) {
+    var entries = [],
+        // jshint validthis:true
+        client = this.client;
+
+    /*jshint -W084 */
+    for(var i= 0,name; name = arguments[i]; ++i) {
+      switch(name) {
+        case 'loader':
+          entries.push(ss.bundler.browserifyLoader());
+          break;
+        case 'libs':
+          entries.push(ss.bundler.systemLibs());
+          break;
+        case 'start':
+          entries = entries.concat(ss.bundler.constants(client));
+          if (client.includes.initCode) {
+            entries = entries.concat(ss.bundler.startCode(client));
+          }
+          break;
 
-      // mixin client constants
-      var client = this.client;
-      if (client.constants) {
-        for(k in client.constants) {
-          result[k] = client.constants[k];
-        }
+        //case "eventemitter2":
+        //case "socketstream":
+        default:
+          entries.push(ss.bundler.systemModule(name));
+          break;
       }
+    }
 
-      return result;
-    },
+    return entries;
+  }
 
-    locals: function() {
-      var result = {}, k;
+  function constants() {
+    var result = {}, k;
 
-      // mixin system locals
-      for(k in system.assets.locals) {
-        result[k] = system.assets.locals[k].value;
-      }
+    // mixin system constants
+    for(k in system.assets.constants) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.